electrum/gui/qt/main_window.py

```diff
@@ -1967,12 +1967,15 @@ def on_pr(self, request: 'paymentrequest.PaymentRequest'):
         else:
             self.payment_request_error_signal.emit()
 
-    def parse_lightning_invoice(self, invoice):
+    def set_ln_invoice(self, invoice: str):
         """Parse ln invoice, and prepare the send tab for it."""
         try:
             lnaddr = lndecode(invoice)
-        except Exception as e:
-            raise LnDecodeException(e) from e
+        except LnInvoiceException as e:
+            self.show_error(_("Error parsing Lightning invoice") + f":\n{e}")
+            return
+
+        self.payto_e.lightning_invoice = invoice
```

Comment:
This line is also done by the caller in paytoedit.py, around line 174. One of them is redundant I guess? Is this one needed?

Reply:
Oh right, don't recall why I added it here. Where do you think it is cleaner to add? I would rather expect it in `set_ln_invoice` in analogy to the android `set_ln_invoice`.

Reply:
Well I suppose if `lndecode` raises, `payto_e.lightning_invoice` should not be set.
That's what the old code is doing at least:
from paytoedit, `parse_lightning_invoice()` is called, and only if it does not raise, `payto_e.lightning_invoice` is set.

The PR is changing the API of `set_ln_invoice`: previously if all was fine it returned None, and if there was a "typical" error, it raised an exception. Now, it returns None in either case.
Look at the caller in paytoedit.py: that code should also be changed probably: it is still expecting the exception for typical errors; and right now with the PR that code will set `payto_e.lightning_invoice` even if it raised (as the caller won't get the exception).

So maybe around L169 in paytoedit.py, this
```
                try:
                    self.win.set_ln_invoice(bolt11_invoice)
                except LnDecodeException as e:
                    self.errors.append(PayToLineError(line_content=data, exc=e))
                else:
                    self.lightning_invoice = bolt11_invoice
```
could be replaced with
```
                self.win.set_ln_invoice(bolt11_invoice)
```

Reply:
Yes, totally makes sense, thank you.